scripts/globals/treasure.lua

```diff
@@ -621,12 +621,8 @@ xi.treasure.treasureInfo =
                 misc =
                 {
                     {
-                        test = function(player) return player:getCharVar("needs_crawler_blood") == 1 end,
-                        code = function(player)
-                            npcUtil.giveKeyItem(player, xi.ki.CRAWLER_BLOOD)
-                            player:setCharVar("needs_crawler_blood", 0)
-                        end,
-                    },
+                        test = function(player) return player:getQuestStatus(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.ENVELOPED_IN_DARKNESS) == QUEST_ACCEPTED and player:getCharVar("Quest[" .. xi.quest.log_id.SANDORIA .. "][" .. xi.quest.id.sandoria.ENVELOPED_IN_DARKNESS .. "]Prog") >= 2 and not player:hasKeyItem(xi.ki.CRAWLER_BLOOD) end,
+                        code = function(player) npcUtil.giveKeyItem(player, xi.ki.CRAWLER_BLOOD) end,                    },
```

Comment:
formatting: In tables please have the closing }, on a new line

{
   stuff
},

Reply:
Resolved